src/main/java/by/exadel/internship/service/impl/InterviewServiceImpl.java

```diff
@@ -99,49 +109,70 @@ public void saveInterviewForForm(UUID formId, InterviewSaveDTO interviewDTO) {
         }
     }
 
-    private void deleteTime(LocalDateTime localDateTime, UUID userId) {
-        UserDTO userDTO = userService.getById(userId);
+    private void saveForm(FormFullDTO formFullDTO){
+        log.info("Try to update Form with uuid = {}", formFullDTO.getId());
+        Form form = formRepository.findByIdAndDeletedFalse(formFullDTO.getId())
+                .orElseThrow(() -> new NotFoundException("Form with uuid = " + formFullDTO.getId() +
+                        " Not Found in DB", "form.uuid.invalid"));
+        form.setFormStatus(formFullDTO.getFormStatus());
+        Interview interview = mapper.toInterview(formFullDTO.getInterview());
+        form.setInterview(interview);
+        formRepository.save(form);
+        log.info("Successfully saved Form with uuid = {} and Interview with uuid = {}",
+                form.getId(),interview.getId());
+    }
+
+    private FormFullDTO getFormById(UUID formId){
+        Form form = formRepository.findByIdAndDeletedFalse(formId).
+                orElseThrow(() -> new NotFoundException("Form with uuid = " + formId +
+                        " Not Found in DB", "form.uuid.invalid"));
+        log.info("Return formFullDTO with uuid = {}", formId);
+        return formMapper.toFormDto(form);
+    }
+
+    private void deleteTime(LocalDateTime localDateTime, UserFullDTO userDTO) {
         List<UserTimeSlotDTO> userTimeSlotDTOList = userDTO.getUserTimeSlots()
                 .stream()
                 .filter(userTimeSlotDTO -> userTimeSlotDTO.getStartDate()
                         .equals(localDateTime)).collect(Collectors.toList());
+        userDTO.getUserTimeSlots().remove(userTimeSlotDTOList.get(0));
```

Comment:
Why are you removing first element?

Reply:
Since first we filter and since user doesn't have a repeating time we will delete the first

Reply:
In this case don't use collect. Use find first instead and if present delete required entity

Reply:
Fixed